app/src/lib/dispatcher/dispatcher.ts

```diff
@@ -848,13 +849,18 @@ export class Dispatcher {
         // this ensures we use the repository root, if it is actually a repository
         // otherwise we consider it an untracked repository
         const path = (await validatedRepositoryPath(action.path)) || action.path
-
         const state = this.appStore.getState()
-        const existingRepository = matchExistingRepository(
+        let existingRepository = matchExistingRepository(
           state.repositories,
           path
         )
 
+        // in case this is valid git repository, there is no need to ask
+        // user for confirmation and it can be added automatically
+        if (!existingRepository && (await isGitRepository(path))) {
```

Comment:
:art: to be a bit more explicit: `existingRepository` -> `existingRepository == null`

Reply:
I'm also fine with making this loop a bit more explicit, for better readability. Here's what I settled on for the flow:

```ts
if (existingRepository == null) {
  const isRepository = await isGitRepository(path)
  if (isRepository) {
    const addedRepositories = await this.addRepositories([path])
    existingRepository = addedRepositories[0]
  }
}
```

Reply:
environment

Reply:
environment

Reply:
Thanks @shiftkey, I've applied your idea.